### ISSUE TITLE
False positive

### ISSUE THREAD

**Issue by uncenter** [csstools/stylelint-prefer-nesting#26]
Apologies, I'm not sure how to describe this issue. Looking at this code sample:

```css
div {
  a {
    color: #fff;
  }

  &:hover a {
    color: #000;
  }
}
```

it is equivalent to:

```css
div a {
  color: #fff;
}
div:hover a {
  color: #000;
}
```

However, this rule suggests it to be:

```css
div {
  a {
    color: #fff;

    &:hover & {
      color: #000;
    }
  }
}
```

Which has the very different functional equivalent of:

```css
div a {
  color: #fff;
}
div a:hover div a {
  color: #000;
}
```

Because of course the `&` part of the original `&hover: a` selector has a different meaning in the new context.

I don't there is a way to simplify the original example at all, so this false positive should just be ruled out.

**Comment by romainmenke** [comment]
This has been fixed in 6.0.2:

https://stylelint.io/demo/#N4Igxg9gJgpiBcIoEsBuACYAdAdu9AhprvvpADYQBO86AxAGZMDcJ6AvrmwGTwAWEVDCqFieUhWq06ABjmtxnHOxAAaEA2TkYAOQIBbOIhgAPAwAdtAOjABnW2vAQcmgOYIQp89QAu6WAwEAK7kftjiWCCWQa7IOLaRtADabPiRtj4AntrkcT4AtEG2MPk4MBlxrpFsALqqbJFUIeWJYqTokXYZEBDktgD0RSVlFThVCB0gBOQA7gSZCSDoAD7Lk8iuONQw1Yq47AqOkC4bAGLU+gQ+HhnZMLk4PjbOblb6AFYO6rDmDojhaRAtxyeVakXIV3KPki9QiQKyIMe+WObnyGQIOCgBCoUDBIAhPihMIa8LuDwKQ1KUMqeIAbFYZFYAEy7JQgdhAA
